### PR TITLE
1.1 filter media

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Media can be filtered on type when displayed from media folder
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/179
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/175

### CODE REVIEW

File: MediaAdminBundle/Resources/public/coffee/MediaChannel.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
+###*
+ * @channel Media
+ * Event available
+ *  - mediasFiltered(mediaCollection) get a media collection filtered
+###
+(($, OpenOrchestra) ->
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/coffee/MediaChannel.coffee]: can you add this file in folder media to respect the namespace

File: MediaAdmin/FileAlternatives/Strategy/AbstractFileAlternativesStrategy.php [hunk 2]
```diff
@@ -10,11 +10,22 @@
...
     protected $tmpDir;
     protected $fileSystem;
 
     /**
+     * Get the $media type supported by the strategy
+     *
+     * @return srtring
+     */
+    public function getMediaType() {
+        return static::MEDIA_TYPE;
```
Comment (alavieille) [inline MediaAdmin/FileAlternatives/Strategy/AbstractFileAlternativesStrategy.php:23]: i think that is not a good idea to specify an abstract media. but just add an abstract method getMediaType to forced an integrator specify a media type

File: MediaAdminBundle/Resources/public/coffee/view/WidgetMediaTypeFilterView.coffee [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+###
+window.OpenOrchestra or= {}
+window.OpenOrchestra.Media or= {}
+
+###*
+ * @class WidgetMediaTypeFilterView
+###
+class OpenOrchestra.Media.WidgetTypeFilterView extends OrchestraView
+  events:
+    'click ul>li>a': 'filterMedia'
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/coffee/view/WidgetMediaTypeFilterView.coffee]: i think it's better to specify a class

File: MediaAdminBundle/Resources/public/coffee/view/WidgetMediaTypeFilterView.coffee [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  ###
+  render: ->
+    viewContext = @
+    $.ajax
+      url: viewContext.options.media_types_url
+      method: 'GET'
+      success: (response) ->
+        viewContext.noFilterUrl = response.links._no_filter
+        viewContext.filters = response.media_types
+        viewContext.drawWidget('')
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/coffee/view/WidgetMediaTypeFilterView.coffee]: i think use a default value for selecteFilter is better and used null not empty string

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 5]
```diff
@@ -129,4 +138,20 @@ public function uploadAction($folderId, Request $request)
...
+
+    /**
+     * @param string folderId
+     *
+     * @Config\Route("/{folderId}/media-types", name="open_orchestra_api_media_type_list")
+     * Config\Method({"POST"})
+     *
+     * @return FacadeInterface|Response
+     */
+    public function mediaTypeListAction($folderId)
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php:150]: may be specify a role for this method

**Comment by ngilain** [comment]
@alavieille : update done

**Comment by alavieille** [comment]
:+1:

**Pull request merged**